lib/src/pubspec.dart

```diff
@@ -158,10 +157,10 @@ class Pubspec {
   }
 }
 
-Version _versionFromString(String input) =>
+Version? _versionFromString(String? input) =>
     input == null ? null : Version.parse(input);
 
-Map<String, VersionConstraint> _environmentMap(Map source) =>
+Map<String, VersionConstraint?>? _environmentMap(Map? source) =>
```

Comment:
is a null constraint actually possible/supported?

Reply:
it ends up mapping to `VersionConstraint.any` – I validate this in tests.

Reply:
I take that back! It seems this is not explicit anywhere that it's required. We could start being strict now, though...

I haven't seen a case where `null` is allowed.

Reply:
Gah! I take that back. My logic here is based on what's allowed by the pub client. The pub client allows null, so I"m going to leave it as-is.